anchore_engine/services/policy_engine/engine/feeds/sync.py

```diff
@@ -120,6 +120,72 @@ def get_selected_feeds_to_sync(config):
         return [VulnerabilityFeed.__feed_name__, NvdV2Feed.__feed_name__]
 
 
+def get_selected_feeds_to_sync(config):
+    """
+    Returns the feeds to be synced based on the configuration.
+    Handles both legacy (top-level) and vulnerabilities provider (policy-engine) configurations
+
+    :param config: dict that is the system configuration
+    :return: list of strings of feed names to sync
+    """
+
+    # check for the vulnerabilities provider section first
+    provider_config = (
+        config.get("services", {}).get("policy_engine", {}).get("vulnerabilities", {})
+    )
+
+    if provider_config:  # new config added for grype integration found
+        logger.debug("Found a vulnerabilities provider configuration")
+        return get_feeds_to_sync_for_vulnerabilities_provider(provider_config)
+    else:  # fall back to older configuration
+        logger.debug("Falling back to legacy feeds configuration")
+        return get_selected_feeds_to_sync_legacy(config)
+
+
+def get_feeds_to_sync_for_vulnerabilities_provider(provider_config):
+    """
+    Expects a vulnerability provider config dictionary and returns the list of data feeds to sync
+
+    Example input:
+    {
+        "provider": "grype",
+        "feeds": {
+            "sync_enabled": True,
+            "ssl_verify": True,
+            "connection_timeout_seconds": 3,
+            "read_timeout_seconds": 60,
+            "data": {
+                "grypedb": {
+                    "enabled": True,
+                    "url": "https://toolbox-data.anchore.io/grype/databases/listing.json",
+                },
+                "packages": {
+                    "enabled": False,
+                    "anonymous_user_username": "anon@ancho.re",
+                    "anonymous_user_password": "pbiU2RYZ2XrmYQ",
+                    "url": "https://ancho.re/v1/service/feeds",
+                    "client_url": "https://ancho.re/v1/account/users",
+                    "token_url": "https://ancho.re/oauth/token",
```

Comment:
Only the url field is required. The others are all unused legacy fields.

Reply:
thanks! will adjust in following work